locale/circuitpython.pot

```diff
@@ -4039,6 +4052,10 @@ msgstr ""
 msgid "source_bitmap must have value_count of 8"
 msgstr ""
 
+#: shared-bindings/memorymap/AddressRange.c
+msgid "start must be an int"
```

Comment:
There should be helpers to verify this already. At minimum let's use the same error string. `%q` can be used in place of "start" in MP printf and passed the qstr.

Reply:
There's not a helper for unsigned int types that I could find. I'd added one in an earlier commit as `mp_obj_get_uint`, but that involves a change to `py` which is undesirable. So, I've moved its code inside `memorymap_addressrange_make_new`. If you've got second thoughts about this approach, it's no trouble to move it back out to `py/obj.c`.

It turns out that the "start must be an int" message can be done away with entirely by attempting object conversion to an int by means of `mp_unary_op`, which I did with this commit.